test/utils/SchemaTester.js

```diff
@@ -0,0 +1,108 @@
+            test("check for error", async () => {
+                const modelObj = {};
+                modelObj[fieldName] = "a".repeat(minimum - 1);
+
+                const model = new Model(modelObj);
+                try {
+                    await model.validate();
+                } catch (err) {
+                    expect(err.errors[fieldName]).toBeDefined();
+                    expect(err.errors[fieldName]).toHaveProperty("kind", "minlength");
+                    expect(err.errors[fieldName]).toHaveProperty("message",
+                        `Path \`${fieldName}\` (\`${"a".repeat(minimum - 1)}\`) is shorter than the minimum allowed length (${minimum}).`
+                    );
+                }
+            });
+            test("check for success", async () => {
+                const modelObj = {};
+                modelObj[fieldName] = "a".repeat(minimum);
+
+                const model = new Model(modelObj);
+                try {
+                    await model.validate();
+                } catch (err) {
+                    expect(err.errors[fieldName]).not.toBeDefined();
+                }
+            });
+
+        });
+    },
+
+    maxLength: (fieldName, maximum) => {
+        describe(`${fieldName} must have a maximum of ${maximum} characters`, () => {
+            test("check for error", async () => {
+                const modelObj = {};
+                modelObj[fieldName] = "a".repeat(maximum + 1);
+
+                const model = new Model(modelObj);
+                try {
+                    await model.validate();
+                } catch (err) {
+                    expect(err.errors[fieldName]).toBeDefined();
+                    expect(err.errors[fieldName]).toHaveProperty("kind", "maxlength");
+                    expect(err.errors[fieldName]).toHaveProperty("message",
+                        `Path \`${fieldName}\` (\`${"a".repeat(maximum + 1)}\`) is longer than the maximum allowed length (${maximum}).`
+                    );
+                }
+            });
+            test("check for success", async () => {
+                const modelObj = {};
+                modelObj[fieldName] = "a".repeat(maximum);
+
+                const model = new Model(modelObj);
+                try {
+                    await model.validate();
+                } catch (err) {
+                    expect(err.errors[fieldName]).not.toBeDefined();
+                }
+            });
+
+        });
+    },
+
+    dateAfterDate: (laterDate, earlierDate) => {
+        describe(`'${earlierDate}' must be after '${laterDate}'`, () => {
+            test("check for error", () => {
+                const modelObj = {};
+                modelObj[earlierDate] =  new Date("02/01/2020");
+                modelObj[laterDate] = new Date("01/01/2020");
+                const model = new Model(modelObj);
+
+                return model.validate((err) => {
```

Comment:
Please use the promise-based version of validate used above, since I think that this one doesn't always fail the tests since jest simply thinks there are no assertions in this test. Additionally, the version used above provides with better debug messages from what I've seen :slightly_smiling_face: